src/view/components/identityWrapper.jsx

```diff
@@ -0,0 +1,151 @@
+                        <WrappedField
+                          data-test-id={`hashEnabled${index}Field`}
+                          name={`identities.${index}.hashEnabled`}
+                          component={Checkbox}
+                          label="Convert ID to sha256 hash"
+                        />
+                      </InfoTipLayout>
+                    </div>
+                    <div className="u-gapTop">
+                      <FieldLabel
+                        labelFor={`authenticatedState${index}Field`}
+                        label="Authenticated State"
+                      />
+                      <div>
+                        <WrappedField
+                          data-test-id={`authenticatedState${index}Field`}
+                          id={`authenticatedState${index}Field`}
+                          name={`identities.${index}.authenticatedState`}
+                          component={Select}
+                          componentClassName="u-fieldLong"
+                          options={authenticatedStateOptions}
+                        />
+                      </div>
+                    </div>
+                    <div className="u-gapTop">
+                      <InfoTipLayout tip="Adobe Experience Platform will use the identity as an identifier to help stitch together more information about that individual. If left unchecked, the identifier within this namespace will still be collected but the ECID will be used as the primary identifier for stitching.">
+                        <WrappedField
+                          data-test-id={`primary${index}Field`}
+                          name={`identities.${index}.primary`}
+                          component={Checkbox}
+                          label="Primary"
+                        />
+                      </InfoTipLayout>
+                    </div>
+                    <div className="u-gapTop">
+                      <Button
+                        data-test-id={`delete${index}Button`}
+                        label="Delete Identity"
+                        icon={<Delete />}
+                        disabled={values.identities.length === 1}
+                        onClick={() => {
+                          arrayHelpers.remove(index);
+                        }}
+                      />
+                      {values.identities.length === 1 ? (
+                        <span className="Note u-gapLeft">
+                          You must have at least one identity to use this
+                          action.
+                        </span>
+                      ) : null}
+                    </div>
+                  </Well>
+                ))}
+              </div>
+            </React.Fragment>
+          );
+        }}
+      />
+    </React.Fragment>
+  );
+}
+
+IdentityWrapper.propTypes = {
+  values: PropTypes.object,
+  initInfo: PropTypes.object
+};
+
+export default IdentityWrapper;
```

Comment:
It's unfortunate that git shows this as a whole new file. Really, it was renamed from customerIdWrapper.jsx and modified. Other than the "customer ID" rename, I also simplified some of the code because the checks like:

```
{((Array.isArray(values.customerIds) &&	
                values.customerIds.length) ||	
                null)
```

were unnecessary since `values.customerIds` (now `values.identities`) is always defined and has at least one item.